File: LeetCode Solutions/Problems/908. Smallest Range I.cpp
class Solution {
public:
    int smallestRangeI(vector<int>& nums, int k) {
        int length = nums.size();
        int minimum, maximum;
        minimum = maximum = nums[0];
        for (int i=1; i<length; i++){
            if (nums[i] > maximum){
                maximum = nums[i];
            }
            else if (nums[i] < minimum){
                minimum = nums[i];
            }
        }
        minimum += min(maximum - minimum, k);
        maximum -= min(maximum - minimum, k);
        return maximum - minimum;
    }
};

File: LeetCode Solutions/Problems/908. Smallest Range I.java
class Solution {
    public int smallestRangeI(int[] nums, int k) {
        int length = nums.length;
        int min, max;
        min = max = nums[0];
        for (int i=1; i<length; i++){
            if (nums[i] > max){
                max = nums[i];
            }
            else if (nums[i] < min){
                min = nums[i];
            }
        }
        min += Math.min(max - min, k);
        max -= Math.min(max - min, k);
        return max - min;
    }
}